apps/arena/lib/arena/matchmaking/quick_game_mode.ex

```diff
@@ -95,18 +99,20 @@ defmodule Arena.Matchmaking.QuickGameMode do
         []
       end
 
-    {:ok, game_pid} =
-      GenServer.start(Arena.GameUpdater, %{
-        clients: clients,
-        bot_clients: bot_clients,
-        game_params: game_params
-      })
+    ## For Battle Royale (quick game mode) there are no teams so we assing each player to a different team
+    {teams, _} =
+      Enum.reduce(clients ++ bot_clients, {[], 1}, fn client, {team_acc, team_id} ->
+        client = Map.put(client, :team, team_id)
+        {[client | team_acc], team_id + 1}
+      end)
```

Comment:
Can we put this team assignment in a separate function? We could create functions for this and we could pattern-match the game mode to assign the teams accordingly.
It will be easier to maintain and also this code block is repeated.

PS: there's a typo and its assign not assing.
```suggestion
    ## For Battle Royale (quick game mode) there are no teams so we assign each player to a different team
    {teams, _} =
      Enum.reduce(clients ++ bot_clients, {[], 1}, fn client, {team_acc, team_id} ->
        client = Map.put(client, :team, team_id)
        {[client | team_acc], team_id + 1}
      end)
```

Reply:
Done, went a bit more and cleaned up `QuickGameMode` duplicated logic by calling directly `GameLauncher` function (essentially they are supposed to be almost the same, so didn't see any harm on it)